### PR TITLE
Not read heaptuple after closing pg_rewrite

### LINKED ISSUE #5252: Memory error in GetDependingView
https://github.com/citusdata/release-test-results/blob/onur-vg-10.2-multi/09_07_2021_1631031248/periodic_job_results/09_07_2021_1631031248/valgrind_test_log_citus.txt

On enterprise-master, https://github.com/citusdata/citus-enterprise/commit/11aa2736cb2956d1860ed83d72ba1525439e44e0:
Using pg-14beta3.
Caught by check-multi-vg.

```c
==20996== VALGRINDERROR-BEGIN
==20996== Invalid read of size 4
==20996==    at 0x1048531B: GetDependingView (home/pguser/citus-enterprise/src/backend/distributed/metadata/dependency.c:1274)
==20996==    by 0x10485D44: BuildViewDependencyGraph (home/pguser/citus-enterprise/src/backend/distributed/metadata/dependency.c:1144)
==20996==    by 0x10485DBB: GetDependingViews (home/pguser/citus-enterprise/src/backend/distributed/metadata/dependency.c:1216)
==20996==    by 0x10434198: GetViewCreationCommandsOfTable (home/pguser/citus-enterprise/src/backend/distributed/commands/alter_table.c:1243)
==20996==    by 0x10434B95: ConvertTable (home/pguser/citus-enterprise/src/backend/distributed/commands/alter_table.c:564)
==20996==    by 0x104354BC: AlterDistributedTable (home/pguser/citus-enterprise/src/backend/distributed/commands/alter_table.c:428)
==20996==    by 0x1043558E: alter_distributed_table (home/pguser/citus-enterprise/src/backend/distributed/commands/alter_table.c:298)
==20996==    by 0x675380: ExecInterpExpr (home/pguser/postgres-source/postgresql-14beta3/src/backend/executor/execExprInterp.c:725)
==20996==    by 0x671C30: ExecInterpExprStillValid (home/pguser/postgres-source/postgresql-14beta3/src/backend/executor/execExprInterp.c:1824)
==20996==    by 0x6AD48F: ExecEvalExprSwitchContext (home/pguser/postgres-source/postgresql-14beta3/src/backend/executor/../../../src/include/executor/executor.h:339)
==20996==    by 0x6AD48F: ExecProject (home/pguser/postgres-source/postgresql-14beta3/src/backend/executor/../../../src/include/executor/executor.h:373)
==20996==    by 0x6AD48F: ExecResult (home/pguser/postgres-source/postgresql-14beta3/src/backend/executor/nodeResult.c:136)
==20996==    by 0x681587: ExecProcNodeFirst (home/pguser/postgres-source/postgresql-14beta3/src/backend/executor/execProcnode.c:463)
==20996==    by 0x679EFC: ExecProcNode (home/pguser/postgres-source/postgresql-14beta3/src/backend/executor/../../../src/include/executor/executor.h:257)
==20996==    by 0x679EFC: ExecutePlan (home/pguser/postgres-source/postgresql-14beta3/src/backend/executor/execMain.c:1551)
==20996==  Address 0x1504e2a4 is in a rw- anonymous segment
==20996== 
==20996== VALGRINDERROR-END
```

### PR DESCRIPTION
**Pull request by onurctirtir**
Fixes #5252.

Should be backported up to and and including 9.5.
Not sure why valgrind couldn't catch that before.

TODO list for myself:
- [ ] Backport to 9.5
- [ ] Backport to 10.0
- [ ] Backport to 10.1

### CODE REVIEW

File: src/backend/distributed/metadata/dependency.c [hunk 1]
```diff
@@ -1218,12 +1218,14 @@ GetDependingView(Form_pg_depend pg_depend)
...
 	bool isDifferentThanRef = pg_rewrite->ev_class != pg_depend->refobjid;
 
-	systable_endscan(rscan);
-	relation_close(rewriteRel, AccessShareLock);
-
+	Oid dependingView = InvalidOid;
 	if ((isView || isMatView) && isDifferentThanRef)
 	{
-		return pg_rewrite->ev_class;
+		dependingView = pg_rewrite->ev_class;
```
Comment (onderkalaci) [inline src/backend/distributed/metadata/dependency.c:1224]: Don't we also need `HeapTupleIsValid(rewriteTup)` before doing all this?

Comment (onurctirtir) [reply src/backend/distributed/metadata/dependency.c:1224]: yeah, definitely, let me do that as well

Comment (onurctirtir) [reply src/backend/distributed/metadata/dependency.c:1234]: done

File: src/backend/distributed/metadata/dependency.c [hunk 2]
```diff
@@ -1212,18 +1212,30 @@ GetDependingView(Form_pg_depend pg_depend)
...
 
 	HeapTuple rewriteTup = systable_getnext(rscan);
+	if (!HeapTupleIsValid(rewriteTup))
+	{
+		/*
+		 * This function already verified that objid's classid is
+		 * RewriteRelationId, so it should exists. But be on the
+		 * safe side.
+		 */
+		elog(ERROR, "catalog lookup failed for function %u", pg_depend->objid);
```
Comment (onderkalaci) [inline src/backend/distributed/metadata/dependency.c:1222]: Maybe use ereport?

**Review by onderkalaci — APPROVED** [review]

**Pull request merged**